### ISSUE TITLE
Evaluation - How to obtain Distance based metric for different scenes in the AiMotive val catalogue?

### ISSUE THREAD

**Issue by PaulSudarshan** [aimotive/mm_training#8]
Please mention on how to obtain distance based metric for different scenes. @TamasMatuszka

**Comment by TamasMatuszka** [comment]
Hi @PaulSudarshan,

You can modify the x_range argument of this [method](https://github.com/aimotive/mm_training/blob/main/exps/mm_training_aim.py#L416 ) to the distance you want to use.

**Comment by PaulSudarshan** [comment]
@TamasMatuszka thanks, and I suppose x_range value should be given in meters? Also can you provide some resources to know about the metrics  used for evaluating the models in the attached image. 
![metrics](https://github.com/aimotive/mm_training/assets/65910042/43cf0768-94c4-4070-9d51-81807436c147)

**Comment by TamasMatuszka** [comment]
@PaulSudarshan Yes, the x_range value is given in meters. You can find some references to the used metrics here:

- AP_AUC and AP_interp: [Section 4.2](http://host.robots.ox.ac.uk/pascal/VOC/pubs/everingham10.pdf) 
- AOS: [Section 2.5](https://www.cvlibs.net/publications/Geiger2012CVPR.pdf)

**Comment by PaulSudarshan** [comment]
> @PaulSudarshan Yes, the x_range value is given in meters. You can find some references to the used metrics here:
> 
> * AP_AUC and AP_interp: [Section 4.2](http://host.robots.ox.ac.uk/pascal/VOC/pubs/everingham10.pdf)
> * AOS: [Section 2.5](https://www.cvlibs.net/publications/Geiger2012CVPR.pdf)

Thanks for providing the resources. Can you clarify few more things : 
1) test_num_preds - does it mean total predicted bounding boxes on the test set?
2) test_precision_2d , what does "2d" mean here?
3) test_precision_op, what does "op" mean here?
4) what does "test_cls_accuracy_bev" mean?

**Comment by TamasMatuszka** [comment]
@PaulSudarshan Sure.

1. Yes, it is the number of predicted bounding boxes.
2. 2D is a heritage of an old code, it used to be utilized for 2D metric calculation. However, the metrics are calculated in BEV space so the values returned by the evaluator are not 2D metrics in image space but rather the BEV metrics.
3. 'op' determines the index of the optimal operating point. It maximizes precision*recall. Please refer to this [method](https://github.com/aimotive/mm_training/blob/main/utils/eval.py#L286) for concrete implementation.
4. 'test_cls_accuracy_bev' refers to the classification accuracy.

**Comment by PaulSudarshan** [comment]
> @PaulSudarshan Sure.
> 
> 1. Yes, it is the number of predicted bounding boxes.
> 2. 2D is a heritage of an old code, it used to be utilized for 2D metric calculation. However, the metrics are calculated in BEV space so the values returned by the evaluator are not 2D metrics in image space but rather the BEV metrics.
> 3. 'op' determines the index of the optimal operating point. It maximizes precision*recall. Please refer to this [method](https://github.com/aimotive/mm_training/blob/main/utils/eval.py#L286) for concrete implementation.
> 4. 'test_cls_accuracy_bev' refers to the classification accuracy.

Thanks, so that means "test_precision_op" and "test_recall_op" determines the precision and recall for the most optimum value of score threshold given by "test_score_op", is that correct?  @TamasMatuszka 

If my understanding is correct, all these metrics are in BEV space right?

**Comment by TamasMatuszka** [comment]
@PaulSudarshan your understanding is correct in both cases.
